src/injected/content/clipboard.js

```diff
@@ -1,29 +1,17 @@
-let textarea;
 let clipboardData;
 
-function init() {
-  textarea = document.createElement('textarea');
-  textarea.style.position = 'absolute';
-  textarea.style.width = 0;
-  textarea.style.height = 0;
-  textarea.style.left = '-20px';
-}
-
 function onCopy(e) {
+  document.removeEventListener("copy", onCopy, true);
+  e.stopImmediatePropagation();
   e.preventDefault();
   const { type, data } = clipboardData;
   e.clipboardData.setData(type || 'text/plain', data);
 }
 
 export default function setClipboard({ type, data }) {
   clipboardData = { type, data };
-  if (!textarea) init();
-  document.addEventListener('copy', onCopy, false);
-  document.documentElement.appendChild(textarea);
-  textarea.focus();
+  document.addEventListener("copy", onCopy, true);
   const ret = document.execCommand('copy', false, null);
```

Comment:
If `copy` event fails to emit, `onCopy` will never be called, and the listener will never be removed from `document`.